### PR TITLE
Shortcode support

### PR DESCRIPTION
**Pull request by Fl0ux**
This new middleware will replace any shortcode configured in asgard.core.shortcodes.php.

Here is an example to show you how it works : 

Example of configuration to put in asgard.core.shortcodes.php

'contact' => [
    'view' => 'contact',
    'data' => []
]

"contact" will be the shortcode to be inserted as "[contact]", in a page content for example.
"view" will be the view to replace the shortcode.
"data" can be used to insert some data into the view.

### CODE REVIEW

**Comment by nWidart** [comment]
Hello,

Thank you for your contribution. 
However it would have been nice to discuss this first.

First of all I'm not really sure what this does based on what you described.

Next please follow the contribution guide and follow PSR1/2, basic on the .php_cs config file present.

File: Http/Middleware/ShortcodeMiddleware.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        $response = $next($request);
+
+        $shotcodes = config('asgard.core.shortcodes');
```
Comment (nWidart) [inline Http/Middleware/ShortcodeMiddleware.php]: Should have a default value as second argument to empty array.

**Comment by nWidart** [comment]
Also: v1.0 is in feature freeze. Meaning new features should go to the 2.0 branches.

**Comment by armababy** [comment]
I guess this allows to do something things like [youtube_video, 'path_to_view_partial', 'youtube link and any data passed to view'], then it filters this and replaces with what view came up with and inserts instead of this shortcode into content.

**Comment by Fl0ux** [comment]
Sorry for such a light description. Actually it simply replace a shortcode like "[youtube]" by a view.

In the configuration file you can define the name of the shortcode, the name of the view to use and you can also add some variables to be injected in that view.

Ex : 

<?php

return [
    'youtube' => [
        'view' => 'myyoutubeview',
        'data' => [
           'name' => 'Robert',
           'test' => 'foobar'
       ]
];

It would be more interesting to do something like "[youtube_video, 'path_to_view_partial', 'youtube link and any data passed to view']" to pass variables. Maybe I'll rewrite my code and submit a new PR for v2 if you are interested.

**Comment by armababy** [comment]
All around could be nice i guess.
You should be able to disable/enable shortcode support trough config or setting.
Potentially could supply CKeditor with shortcode list etc.

I can see use for this. A lot easier to import pre-made view with tabular data for example than fighting with editor.

**Pull request closed without merging**